test/util/genesis/genesis.go

```diff
@@ -222,10 +244,45 @@ func (g *Genesis) Export() (*coretypes.GenesisDoc, error) {
 		gentxs,
 		g.accounts,
 		g.GenesisTime,
-		g.genOps...,
 	)
 }
 
+// ExportBytes generates and returns a serialized genesis document as raw bytes that can be written to a file.
+func (g *Genesis) ExportBytes() ([]byte, error) {
+	gentxs, err := g.getGenTxs()
+	if err != nil {
+		return nil, err
+	}
+
+	switch g.appVersion {
+	// app versions 1, 2 and 3 are all handled with in app logic in V3.
+	case 1, 2, 3:
+		return DocumentLegacyBytes(
+			loadV3GenesisAppState(),
+			g.ecfg,
+			g.ConsensusParams,
+			g.ChainID,
+			gentxs,
+			g.accounts,
+			g.GenesisTime,
+		)
+	case 4:
+		tempApp := app.New(log.NewNopLogger(), dbm.NewMemDB(), nil, 0, simtestutil.EmptyAppOptions{})
```

Comment:
we can only use the `tempApp` approach for v4, as it is a v4 app, the v3 genesis file looks slightly different and so we need to use an alternate approach to produce it.